Description: A recent fix caused a regression which makes
 memory allocation errors pop up when killing the program
 or when it stops because of an error. 
Author: Ruben Undheim <ruben.undheim@gmail.com>
Forwarded: Fixed upstream - not released yet
Applied-Upstream: 0ac725b7c56e43434b8b09e2793b5c93e1b79ce7
---
Index: mrtdreader/src/mrtdreader.c
===================================================================
--- mrtdreader.orig/src/mrtdreader.c	2014-12-19 19:56:13.358746802 +0100
+++ mrtdreader/src/mrtdreader.c	2014-12-19 19:56:44.378747066 +0100
@@ -138,12 +138,18 @@
 		mrtd_bac_get_kmrz(pn, dob, eov, buffer);
 		kmrz = buffer;
 	}
-	if(pn != NULL)
+	if(pn != NULL) {
 		free(pn);
-	if(dob != NULL)
+		pn = NULL;
+	}
+	if(dob != NULL){
 		free(dob);
-	if(eov != NULL)
+		dob = NULL;
+	}
+	if(eov != NULL){
 		free(eov);
+		eov = NULL;
+	}
 
 
 	signal(SIGINT, forcestop);
